Tasks/Maven/maventask.ts

```diff
@@ -3,97 +3,181 @@
 import tl = require('vsts-task-lib/task');
 import path = require('path');
 
-var mvntool = '';
-var mavenVersionSelection = tl.getInput('mavenVersionSelection', true);
+var mavenPOMFile: string = tl.getPathInput('mavenPOMFile', true, true);
+var javaHomeSelection: string = tl.getInput('javaHomeSelection', true);
+var mavenVersionSelection: string = tl.getInput('mavenVersionSelection', true);
+var mavenGoals: string[] = tl.getDelimitedInput('goals', ' ', true); // This assumes that goals cannot contain spaces
+var mavenOptions: string = tl.getInput('options', false); // Options can have spaces and quotes so we need to treat this as one string and not try to parse it
+var publishJUnitResults: string = tl.getInput('publishJUnitResults');
+var testResultsFiles: string = tl.getInput('testResultsFiles', true);
+
+// Determine the version and path of Maven to use
+var mvnExec: string = '';
 if (mavenVersionSelection == 'Path') {
-    tl.debug('Using maven path from user input');
+    // The path to Maven has been explicitly specified
+    tl.debug('Using Maven path from user input');
     var mavenPath = tl.getPathInput('mavenPath', true, true);
-    mvntool = path.join(mavenPath, 'bin/mvn');
+    mvnExec = path.join(mavenPath, 'bin/mvn');
 
+    // Set the M2_HOME variable to a custom Maven installation path?
     if (tl.getBoolInput('mavenSetM2Home')) {
-        tl.setEnvVar("M2_HOME", mavenPath);
+        tl.setEnvVar('M2_HOME', mavenPath);
         tl.debug('M2_HOME set to ' + mavenPath)
```

Comment:
The lib should output debug for you.  Confirm.  That's the point of the lib - debug for free.